### PR TITLE
VITIS-5805 QDMA: XRT to support pre downloaded QDMA lib driver files using XRT build options

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Signed-off-by: Rajkumar Rampelli <rajkumar@xilinx.com>

#### Problem solved by the commit
Support pre downloaded QDMA drivers to build it in XRT. 
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
VITIS-5805
#### How problem was solved, alternative solutions (if any) and why they were rejected
Introduced new build option -qdmalib to provide pre downloaded QDMA driver path to XRT
#### Risks (if any) associated the changes in the commit
NA
#### What has been tested and how, request additional testing if necessary
XRT build with new introduced option
bash-4.2# ./build.sh -qdmalib /scratch/dma_ip_drivers/
#### Documentation impact (if any)
NA

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
Tested this PR using below steps

bash-4.2# git clone https://github.com/rajkumar-xilinx/XRT.git -b 11may
Cloning into 'XRT'...
remote: Enumerating objects: 87791, done.
remote: Counting objects: 100% (439/439), done.
remote: Compressing objects: 100% (258/258), done.
remote: Total 87791 (delta 270), reused 314 (delta 181), pack-reused 87352
Receiving objects: 100% (87791/87791), 72.99 MiB | 10.82 MiB/s, done.
Resolving deltas: 100% (69787/69787), done.
bash-4.2# cd XRT/
bash-4.2# ls src/runtime_src/core/pcie/driver/linux/xocl/lib/libqdma/
bash-4.2# cd build/
bash-4.2# ./build.sh -opt -noctest -qdmalib /home/rahul/raj/dma_ip_drivers/
bash-4.2# ls ../src/runtime_src/core/pcie/driver/linux/xocl/lib/libqdma/
docs  index.html  QDMA  README.md  XDMA  XVSEC

**Comment by chvamshi-xilinx** [comment]
Hi @rajkumar-xilinx , Can you please update the description more explicit about offline installation etc.

**Comment by rajkumar-xilinx** [comment]
I see below error from the build log: I will fix and I will add reviewers again.
> 00:46:49 Successfully installed pybind11-2.9.2
> 00:46:50 fatal: Needed a single revision
> 00:46:50 Unable to find current revision in submodule path 'src/runtime_src/core/pcie/driver/linux/xocl/lib/libqdma'

**Comment by chvamshi-xilinx** [comment]
retest this please

File: src/runtime_src/core/pcie/driver/linux/CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
+)
+
+function(CopyQDMALIB)
+ message(STATUS "Copying QDMA lib file from $ENV{QDMA_LIB_DIR}")
+ file(COPY $ENV{QDMA_LIB_DIR} DESTINATION ${QDMA_LIB_DESTINATION_DIR})
+endfunction()
+
+if(DEFINED ENV{QDMA_LIB_DIR})
+ CopyQDMALIB()
+endif()
```
Comment (stsoe) [inline src/runtime_src/core/pcie/driver/linux/CMakeLists.txt:21]: Shouldn't this be implemented as a target dependency?   Now copying will be a CMake runtime, but it should be  a dependency based copying and what ever target requires libqdma should trigger the copying. 

Also please implement the env as a CMake variable and add it to cmake_flags in build.sh.

**Review by stsoe — CHANGES_REQUESTED** [review]

**Pull request closed without merging**